### PR TITLE
Fixed results and variables storage tables definitions and UUID PK insertion

### PR DESCRIPTION
**Pull request by julien-sebire**
Draft PR since it has to be reviewed then required in package-tao composer.json for deployment, but not merged to develop until the benchmark is run. The version bump will be done then.

### CODE REVIEW

File: model/RdsResultStorage.php [hunk 1]
```diff
@@ -638,6 +643,16 @@ protected function getResultRow($variable)
...
     }
 
+    /**
+     * Generates a unique, not auto-increment based, primary key.
+     *
+     * @return string
+     */
+    public function getUniquePrimaryKey()
+    {
+        return strrev(uniqid('', true));
```
Comment (SergiiTao) [inline model/RdsResultStorage.php]: does strrev make uniqid more unique? :)

**Comment by tikhanovichA** [comment]
And what about the unit test 'RdsResultStorageTest'? As I see, it's broken.

File: model/RdsResultStorage.php (hunk 1 above)
Comment (julien-sebire) [reply model/RdsResultStorage.php]: The use of strrev allows use to have non-consecutive uuids (base on currenttime) to be given to spanner as primary key of the table, on which spanner bases the spreading over nodes.
Without the strrev, we would end up with sequenced PK and all would be saved to the same spanner node, creating hotspot and loosing the sharding feature of spanner.

File: scripts/install/createTables.php [hunk 2]
```diff
@@ -48,24 +48,22 @@ public function generateTables(\common_persistence_SqlPersistence $persistence)
...
+            $tableVariables = $schema->createtable(RdsResultStorage::VARIABLES_TABLENAME);
+            $tableVariables->addColumn(RdsResultStorage::VARIABLES_TABLE_ID, 'string', ['length' => 25, 'notnull' => true]);
+            $tableVariables->addColumn(RdsResultStorage::CALL_ID_TEST_COLUMN, 'string', ['length' => 255]);
+            $tableVariables->addColumn(RdsResultStorage::CALL_ID_ITEM_COLUMN, 'string', ['length' => 255]);
+            $tableVariables->addColumn(RdsResultStorage::TEST_COLUMN, 'string', ['length' => 255]);
+            $tableVariables->addColumn(RdsResultStorage::ITEM_COLUMN, 'string', ['length' => 255]);
+            $tableVariables->addColumn(RdsResultStorage::VARIABLE_VALUE, 'text', []);
+            $tableVariables->addColumn(RdsResultStorage::VARIABLE_IDENTIFIER, 'string', ['length' => 255]);
+            $tableVariables->addColumn(RdsResultStorage::VARIABLES_FK_COLUMN, 'string', ['length' => 255]);
+            $tableVariables->addColumn(RdsResultStorage::CREATED_AT, 'string', ['length' => 255]);
```
[the remark this answers is not in the document]
Comment (tikhanovichA) [reply scripts/install/createTables.php]: and do we need '['notnull' => true]' for this one?

**Comment by julien-sebire** [comment]
Right, my bad. Default value for `notnull` is `true`, not `false`.
Fixing.

**Comment by julien-sebire** [comment]
Unit test fixed also.

File: model/RdsResultStorage.php (hunk 1 above)
Comment (ekkinox) [reply model/RdsResultStorage.php]: applies to bigtable btw also

File: model/RdsResultStorage.php [hunk 3]
```diff
@@ -275,7 +276,7 @@ public function getVariables($callId)
             ->from(self::VARIABLES_TABLENAME)
             ->andWhere(self::CALL_ID_ITEM_COLUMN .' IN(:ids)')
             ->orWhere(self::CALL_ID_TEST_COLUMN .' IN(:ids)')
-            ->orderBy(self::VARIABLES_TABLE_ID)
+            ->orderBy(self::CREATED_AT)
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/RdsResultStorage.php]: will this work as expected on GC Spanner?
does the newSQL handle properly the order by ?

File: scripts/install/createTables.php [hunk 4]
```diff
@@ -58,14 +58,15 @@ public function generateTables(\common_persistence_SqlPersistence $persistence)
             $tableResults->setPrimaryKey(array(RdsResultStorage::RESULTS_TABLE_ID));
 
 
-            $tableVariables->addColumn(RdsResultStorage::VARIABLES_TABLE_ID, "integer", array("autoincrement" => true));
+            $tableVariables->addColumn(RdsResultStorage::VARIABLES_TABLE_ID, 'string', ['length' => 25]);
```
Comment (ekkinox) [inline scripts/install/createTables.php]: uniqid('', true) length = 23, maybe we can spare those 2 ?

File: model/RdsResultStorage.php (hunk 3 above)
Comment (julien-sebire) [reply model/RdsResultStorage.php]: Yes, this is DBAL generating a SQL statement with plain "ORDER BY created_at" clause, and Spanner executes native SQL queries.

File: scripts/install/createTables.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (julien-sebire) [reply scripts/install/createTables.php]: Fixed

File: scripts/install/createTables.php (hunk 4 above)
Comment (julien-sebire) [reply scripts/install/createTables.php]: Fixed with 23 bytes.

**Comment by julien-sebire** [comment]
This PR is replaced by https://github.com/oat-sa/extension-tao-outcomerds/pull/90.

**Pull request closed without merging**